facebook/OpenBIC
# Artemis: cb: Support switch PLDM firmware update

# Description
- Support switch PLDM firmware update
- Support to read switch firmware update via PLDM

# Motivation
- Support switch PLDM firmware update
- Support to read switch firmware update via PLDM

# Test Plan
- Build code: Pass
- Get switch firmware: Pass
- PLDM update switch firmware: Pass
- Check switch firmware cache: Pass

# Log
root@bmc-oob:~# fw-util cb --version
CB BIC Version: cb2023.35.e1
CB CPLD Version: 00040e12
CB PESW0 Version: 52.d7.00.07
CB PESW1 Version: 52.d7.01.07
CB PESW_VR Version: Infineon 49f0fe5c, Remaining Write: 17
root@bmc-oob:~# fw-util cb --update pesw0 at_cb_pex89000_v52.D7.00.08_pex0.pldm
RequestUpdate Success.
PassComponentTable Success.
UpdateComponent Success.
Download offset : 0x00150b00/0x00150b64, size : 0x00000064
TransferComplete.
VerifyComplete.
ApplyComplete.
ActivateFirmwareComplete.
Upgrade of cb : pesw0 succeeded
root@bmc-oob:~# fw-util cb --update pesw1 at_cb_pex89000_v52.D7.01.08_pex1.pldm
RequestUpdate Success.
PassComponentTable Success.
UpdateComponent Success.
Download offset : 0x00150b00/0x00150b64, size : 0x00000064
TransferComplete.
VerifyComplete.
ApplyComplete.
ActivateFirmwareComplete.
Upgrade of cb : pesw1 succeeded
root@bmc-oob:~# power-util mb cycle
Power cycling fru 1...
root@bmc-oob:~# power-util mb status
Power status for fru 1 : ON
root@bmc-oob:~# fw-util cb --version
CB BIC Version: cb2023.35.e1
CB CPLD Version: 00040e12
CB PESW0 Version: 52.d7.00.08
CB PESW1 Version: 52.d7.01.08
CB PESW_VR Version: Infineon 49f0fe5c, Remaining Write: 17

root@bmc-oob:~# cat /tmp/cache_store/cb_pex0_active_ver
52.d7.00.08root@bmc-oob:~#
root@bmc-oob:~# cat /tmp/cache_store/cb_pex1_active_ver
52.d7.01.08root@bmc-oob:~#
